Switch replicator manager change feeds to "longpoll"

nickva opened the pull request:
Fixes replication manager rescans on cluster membership
change.

Replication manager resets all replication db
sequence checkpoints, and starts a new replicator db
background scanner.  Each replicator database is signaled
to rescan from sequence 0. However previous change feeds
for each db have to exit first. If they never exit, because
they are "continuous" new change feeds will never start.

Putting change feeds in "longpoll" mode ensures they will
eventually exit.

JIRA: COUCHDB-2963

davisp commented:
+1 I'd also note that the fact its continuous in the first place looks to be a bad merge from when we ported the clustered version of the replicator back to the open source version. So this is mostly just clearing up a case that we had fixed but accidentally broke during that work.

davisp commented:
I wouldn't bother adding that to the commit message. That's just for a historical note to explain the background of the PR. The commit message is clear enough on its own for the specific change.

kxepal commented:
How they'll eventually exit if timeout is infinity? IIRC longpoll feeds has the only special option to early break when they hit the limit of requested changes, but it's not applicable here.

nickva commented:
@kxepal

Longpolling works because we start the change feed after db has notified that it has changes, so we ask it for all changes since checkpointed sequence. So change feeds yield changes since last checkpoint and then stop.

kxepal commented:
Right, I forgot about this moment and get confused by timeout. Thanks!
